controller/konnect/ops/ops_controlplane.go

```diff
@@ -202,5 +232,20 @@ func getControlPlaneForUID(
 		return "", fmt.Errorf("failed listing %s: %w", cp.GetTypeName(), ErrNilResponse)
 	}
 
-	return getMatchingEntryFromListResponseData(sliceToEntityWithIDSlice(resp.ListControlPlanesResponse.Data), cp)
+	id, err := getMatchingEntryFromListResponseData(sliceToEntityWithIDSlice(resp.ListControlPlanesResponse.Data), cp)
+	if err != nil {
+		return "", err
+	}
+
+	if err := setGroupMembers(ctx, cl, cp, id, sdkGroups); err != nil {
+		// If we failed to set group membership, we should return a specific error with a reason
+		// so the downstream can handle it properly.
+		return id, KonnectEntityCreatedButRelationsFailedError{
+			KonnectID: id,
+			Err:       err,
+			Reason:    konnectv1alpha1.KonnectGatewayControlPlaneProgrammedReasonFailedToSetControlPlaneGroupMembers,
+		}
+	}
+
+	return id, nil
```

Comment:
Do you mind adding a test for this as well? Ideally extending the test suite in https://github.com/Kong/gateway-operator/blob/8c7fbd40e945bb95c5247288131eac3df9cc7113/test/envtest/konnect_entities_gatewaycontrolplane_test.go.

There's already 1 testcase for conflict https://github.com/Kong/gateway-operator/blob/8c7fbd40e945bb95c5247288131eac3df9cc7113/test/envtest/konnect_entities_gatewaycontrolplane_test.go#L375-L450. 1 more for CP group with members would be ideal.

Reply:
Done.